### PR TITLE
Needed for for ehcache/ehcache3#1008

### PR DESCRIPTION
**Pull request by mathieucarbou**
Required by ehcache/ehcache3#1008

I had to upgrade some maven dependencies: I ran into some packaging issues and bugs in the maven-jar-plugin that was including classes from dependency module...

the last commit (fd85289) is a refactor for management providers and provides some classes ot facilitate implementation in the context of a server entity

I am adding also some notifications:

- __ENTITY_REGISTRY_AVAILABLE__
- ENTITY_REGISTRY_UPDATED
- __CLIENT_REGISTRY_AVAILABLE__
- CLIENT_REGISTRY_UPDATED

There is a lot of *_UPDATED events. So it could be better to just listen to AVAILABLE and then do the stats updates on other events than REGISTRY_UPDATED since it can be updated but not for the stats

Will need a __5.0.8.beta3__ release also.

### CODE REVIEW

**Comment by mathieucarbou** [comment]
@anthonydahanne : __PLEASE DO NOT MERGE YET__: i am testing that within Ehcache with Galvan and I still have some packaging / classloading issues...

**Comment by anthonydahanne** [comment]
LGTM

**Comment by mathieucarbou** [comment]
@ljacomet : when this PR will be reviewed, released and merged, you'll have access to the ubder jars:

```
org.terracotta.management.dist:management-common
org.terracotta.management.dist:management-client
org.terracotta.management.dist:management-server
```

I am using them for the Ehcache integration tests in my PR: ehcache/ehcache3#1555

**Comment by mathieucarbou** [comment]
@anthonydahanne : ready for review, merge and release: __5.0.8.beta3__

File: management/cluster-topology/src/main/java/org/terracotta/management/model/cluster/ServerEntity.java [hunk 1]
```diff
@@ -25,14 +25,15 @@
  */
 public final class ServerEntity extends AbstractNode<Server> {
 
-  private static final long serialVersionUID = 2;
```
Comment (anthonydahanne) [inline management/cluster-topology/src/main/java/org/terracotta/management/model/cluster/ServerEntity.java]: comment about the commit message (typo) : should read "Add consumerId field on server entity"

Comment (mathieucarbou) [inline management/cluster-topology/src/main/java/org/terracotta/management/model/cluster/ServerEntity.java]: ok I'll reword

**Comment by anthonydahanne** [comment]
LGTM, tested from the TMS, no regression issues ; could see the 4 new capabilities just fine

**Pull request merged**